topi/python/topi/nn/fifo_buffer.py

```diff
@@ -0,0 +1,127 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""FIFO buffer op"""
+from __future__ import absolute_import as _abs
+import tvm
+from .. import tag
+from ..transform import concatenate, strided_slice
+
+@tvm.tag_scope(tag=tag.INJECTIVE+",fifo_buffer")
+def fifo_buffer(data, buffer, axis):
```

Comment:
Do we want to add it into from_mxnet, or wait until ring buffer is upstreamed to mxnet?

Reply:
@kevinthesun Actually, I'm brainstorming a new custom pass where I can annotate a stack of conv so that buffer op can be automatically inserted. So
```
%1 = conv(input_window)   # e.g. (1, 1, 1, 10) -> (1, 1, 1, 8)
%2 = conv(%1)  # e.g. (1, 1, 1, 8) -> (1, 1, 1, 6)
```
would become
```
%1 = fifo_buffer(incremental_input, buffer)   # e.g. (1, 1, 1, 2), (1, 1, 1, 6) -> (1, 1, 1, 6)
%2 = conv(%1)   # e.g. (1, 1, 1, 6) -> (1, 1, 1, 4)
%3 = fifo_buffer(%2, buffer2)  # e.g. (1, 1, 1, 4), (1, 1, 1, 4) -> (1, 1, 1, 4)
%4 = conv(%3)   # e.g. (1, 1, 1, 4) -> (1, 1, 1, 2)
%5 = fifo_buffer(%4, buffer3)  # e.g. (1, 1, 1, 2), (1, 1, 1, 6) -> (1, 1, 1, 6)

# %1 and %3 would be fed back into buffer and buffer2 in the next invocation
```

Reply:
I had an offline discussion with @kevinthesun. He told me that a new compiler pass will bring significant amount of complexity and maintenance burden and would also confuse users. I will plan to submit the operator to MXNet as planned.